File: w07/d03/Sophie_Laffont/hangman.js
function Hangman(word){

  this.progress = function(){
      var guesses = [];
    word.split("").forEach(function(letter){
      guesses.push("_");
    });

    return guesses.join(" ");
  }

  this.guess = function(letter){

   };
}

var game = new Hangman("detective");

console.log(game.progress());
game.guess("e");
console.log(game.progress());

File: w10/d01/Jonathan_Milgrom/hangman/lib/hangman.rb
class Hangman

	attr_accessor :guesses

	def initialize(word)
		@num = Random.new(2)
		if word.class == Array
			@word = word[@num.rand(0..word.length).floor].split("")
		else
			@word = word.split("")
		end
		@progress = @word.map { "_" }
		@incorrect_guesses = 0
		@guesses = []
	end

	def progress
		@incorrect_guesses < 6 ? @progress.join(" ") : @progress = @word.join
	end

	def guess(guess_letter)
		if @incorrect_guesses < 5
			@guesses << guess_letter
			if @word.include?(guess_letter)
				@word.each_with_index do |letter, index|
					if letter == guess_letter
						@progress[index] = letter
					end
				end
				return true
			else
				@incorrect_guesses += 1
				return false
			end
		else 
			@incorrect_guesses = 6
			return nil
		end
	end

	def incorrect
		return @incorrect_guesses
	end

	def giveup
		@incorrect_guesses = 6
		return @progress = @word.join
	end

end